shanoir-ng-datasets/src/main/java/org/shanoir/ng/importer/service/DicomSRImporterService.java

```diff
@@ -0,0 +1,399 @@
+				for (Dataset dataset : datasets) {
+					List<DatasetExpression> expressions = dataset.getDatasetExpressions();
+					for (DatasetExpression expression : expressions) {
+						// only DICOM is of interest here
+						if (expression.getDatasetExpressionFormat().equals(DatasetExpressionFormat.DICOM)) {
+							List<DatasetFile> files = expression.getDatasetFiles();
+							for (DatasetFile file : files) {
+								if (file.isPacs()) {
+									String path = file.getPath();
+									if (path.contains(studyInstanceUID) && path.contains(seriesInstanceUID) && path.contains(sOPInstanceUID)) {
+										return dataset;
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+		LOG.error("Error: dataset could not be found with UIDs from DICOM SR.");	
+		return null;
+	}
+	
+	/**
+	 * Create the dataset in the database.
+	 * 
+	 * @param examination
+	 * @param dataset
+	 * @param datasetAttributes
+	 * @throws MalformedURLException
+	 */
+	private void createDataset(Examination examination, Dataset dataset, Attributes datasetAttributes) throws MalformedURLException {
+		MeasurementDataset measurementDataset = new MeasurementDataset();
+		measurementDataset.setReferencedDatasetForSuperimposition(dataset); // keep link to original dataset
+		measurementDataset.setStudyId(examination.getStudyId());
+		measurementDataset.setSubjectId(examination.getSubjectId());
+		measurementDataset.setCreationDate(LocalDate.now());
+//		completeDatasetFromDicomSR(datasetAttributes, measurementDataset);
+		createMetadata(measurementDataset);
+		createDatasetExpression(datasetAttributes, measurementDataset);
+		datasetService.create(measurementDataset);
+	}
+
+	/**
+	 * Create the dataset metadata.
+	 * 
+	 * @param measurementDataset
+	 */
+	private void createMetadata(MeasurementDataset measurementDataset) {
+		DatasetMetadata originMetadata = new DatasetMetadata();
+		originMetadata.setName(IMAGING_MEASUREMENT_REPORT);
+		originMetadata.setComment(IMAGING_MEASUREMENT_REPORT);
+		originMetadata.setDatasetModalityType(DatasetModalityType.GENERIC_DATASET);
+		originMetadata.setCardinalityOfRelatedSubjects(CardinalityOfRelatedSubjects.SINGLE_SUBJECT_DATASET);		
+		measurementDataset.setOriginMetadata(originMetadata);
+		measurementDataset.setUpdatedMetadata(originMetadata);
+	}
+
+	/**
+	 * This method extracts important data, that could later be stored even
+	 * inside shanoir-database, to optimize mass data export of all measurements
+	 * e.g. for one study. This code works and has been tested to access the most
+	 * important measurement attributes and was hard to construct. Even if currently
+	 * not used, we keep it for a very high later usage.
+	 * 
+	 * @param datasetAttributes
+	 * @param measurementDataset
+	 */
+	private void completeDatasetFromDicomSR(Attributes datasetAttributes, MeasurementDataset measurementDataset) {
+		Sequence contentSequence = datasetAttributes.getSequence(Tag.ContentSequence);
```

Comment:
We can discuss about this but we can imagine having a little bigger if
```
if (a != null && a.get(0) && a.get(0).getAttribute() != null [...]) {
    Attribute att = a.get(0).getAttribute()
}
```
I don't really know if it's better or not.
Otherwise 
```
if (a == null) {
return null;
}
```

Reply:
Hi Jean-Come, currently the above code is not used, see my comment, and only kept in case, we would like to go further on the actual annotation content. On the other hand, even if the if block is unfortunately very deep (related to the weird access to the sub-items in dicom api), I have fear a more compact kind of code would make it even more complicated to understand on which level of item-depth we are operating to access information. I hope it is fine like this. With kind regards, Michael